### PR TITLE
feat: add env to choose which node to decom

### PR DESCRIPTION
**Pull request by jiuker**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description

now is always choose the first pool first endpoint to decom

but if someone have a broken endpoint, the user need to fix that node to keep decom go on always.

## Types of changes
- [x] Bug fix (19001)https://github.com/minio/minio/issues/19001
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by shtripat** [comment]
@jiuker can you please add verification steps for PR?
Is it like bring a distributed multi node, multi pool setup and bring down the first node of first pool, and then try decom operation?

**Comment by jiuker** [comment]
> @jiuker can you please add verification steps for PR? Is it like bring a distributed multi node, multi pool setup and bring down the first node of first pool, and then try decom operation?

1. Deploy minio with two pools with two nodes. (Env:  _MINIO_DECOM_ENDPOINT: The second endpoint.)
2. mc alias set minio firstEndPoint user password.
3. use console to create bucket and upload files. 
4. mc admin decommission minio start firstCmd.

Without this pr The decommission will start at first pool and first endpoint.
This pr is made that you can set the second endpoint to do the decommission for if first endpoint have a broken status like the issue happened.
@shtripat

File: cmd/admin-handlers-pools.go [hunk 1]
```diff
@@ -391,3 +376,24 @@ func (a adminAPIHandlers) RebalanceStop(w http.ResponseWriter, r *http.Request)
 	writeSuccessResponseHeadersOnly(w)
 	logger.LogIf(ctx, pools.saveRebalanceStats(GlobalContext, 0, rebalSaveStoppedAt))
 }
+
+func proxyDecommissionRequest(ctx context.Context, defaultEndPoint Endpoint, w http.ResponseWriter, r *http.Request) (proxy bool) {
+	ep := os.Getenv("_MINIO_DECOM_ENDPOINT")
```
Comment (harshavardhana) [inline cmd/admin-handlers-pools.go]: This is temporary for now - we will add a more official one later.
```suggestion
	host := env.Get("_MINIO_DECOM_ENDPOINT_HOST", "")
```

File: cmd/admin-handlers-pools.go [hunk 2]
```diff
@@ -391,3 +376,24 @@ func (a adminAPIHandlers) RebalanceStop(w http.ResponseWriter, r *http.Request)
...
+	if ep != "" {
+		for nodeIdx, proxyEp := range globalProxyEndpoints {
+			if proxyEp.Endpoint.String() == ep && !proxyEp.IsLocal {
```
Comment (harshavardhana) [inline cmd/admin-handlers-pools.go]:
```suggestion
			if proxyEp.Endpoint.Host == host && !proxyEp.IsLocal {
```

File: cmd/admin-handlers-pools.go [hunk 3]
```diff
@@ -391,3 +376,24 @@ func (a adminAPIHandlers) RebalanceStop(w http.ResponseWriter, r *http.Request)
 	writeSuccessResponseHeadersOnly(w)
 	logger.LogIf(ctx, pools.saveRebalanceStats(GlobalContext, 0, rebalSaveStoppedAt))
 }
+
+func proxyDecommissionRequest(ctx context.Context, defaultEndPoint Endpoint, w http.ResponseWriter, r *http.Request) (proxy bool) {
+	ep := os.Getenv("_MINIO_DECOM_ENDPOINT")
+	if ep != "" {
```
Comment (harshavardhana) [inline cmd/admin-handlers-pools.go]:
```suggestion
	if host != "" {
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/admin-handlers-pools.go [hunk 4]
```diff
@@ -391,3 +376,24 @@ func (a adminAPIHandlers) RebalanceStop(w http.ResponseWriter, r *http.Request)
...
+	}
+	for nodeIdx, proxyEp := range globalProxyEndpoints {
+		if proxyEp.Endpoint.Host == defaultEndPoint.Host && !proxyEp.IsLocal {
+			if proxyRequestByNodeIndex(ctx, w, r, nodeIdx) {
+				return true
+			}
+		}
+	}
+	return false
+}
```
Comment (harshavardhana) [inline cmd/admin-handlers-pools.go]:
```suggestion
	host := env.Get("_MINIO_DECOM_ENDPOINT_HOST", defaultEndPoint.Host)
	if host == "" {
	        return false
	}
	for nodeIdx, proxyEp := range globalProxyEndpoints {
		if proxyEp.Endpoint.Host == host && !proxyEp.IsLocal {
			if proxyRequestByNodeIndex(ctx, w, r, nodeIdx) {
				return true
			}
		}
	}
	return false
}
```

**Review by harshavardhana — COMMENTED** [review]
A bit of simplification

**Review by harshavardhana — APPROVED** [review]

**Comment by broferek** [comment]
Hi @jiuker @harshavardhana 

I have upgraded my cluster to the latest release which includes this PR.

I'm trying testing this new env var but I'm struggling with it.

What should be the expected value of the new var _MINIO_DECOM_ENDPOINT_HOST ?

Say I have 2 pools like dc1_node{1...8} and dc2_node{1...16}.

Without setting the var, dc1_node1 is chosen by default to be the responsible of the decommission.

So if I want another host to do the decom, should I set the var to :
- dc2_node1
- https://dc2_node1:9000
- dc2_node{1...16}
- etc ...

Thanks for your answer

**Comment by jiuker** [comment]
Just host. @broferek

**Pull request merged**